amethyst/src/main/java/com/vitorpamplona/amethyst/ui/components/ShareHelper.kt

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2024 Vitor Pamplona
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.vitorpamplona.amethyst.ui.components
+
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import androidx.core.content.FileProvider
+import com.vitorpamplona.amethyst.Amethyst
+import java.io.File
+import java.io.FileInputStream
+import java.io.IOException
+
+object ShareHelper {
+    private const val TAG = "ShareHelper"
+    private const val DEFAULT_EXTENSION = "jpg"
+    private const val SHARED_FILE_PREFIX = "shared_media"
+
+    // Media type magic numbers
+    private val JPEG_MAGIC = byteArrayOf(0xFF.toByte(), 0xD8.toByte())
+    private val PNG_MAGIC = byteArrayOf(0x89.toByte(), 0x50.toByte(), 0x4E.toByte(), 0x47.toByte())
+    private val WEBP_HEADER_START = "RIFF".toByteArray()
+    private val WEBP_HEADER_END = "WEBP".toByteArray()
+    private val GIF_MAGIC = "GIF8".toByteArray()
+
+    fun getSharableUriFromUrl(
+        context: Context,
+        imageUrl: String,
+    ): Pair<Uri, String> {
```

Comment:
You could make this function `suspend`, and modify this line to:
```suggestion
    ): Pair<Uri, String>  =  withContext(Dispatchers.IO){
```
Using Dispatchers.IO here, since we are dealing with file manipulation.